### DISCUSSION TITLE
15/07/2023 Meeting Topics

### DISCUSSION THREAD

**Discussion by mindoodoo** [ratatui/ratatui#313] [Discussion]
Hey all, a public meeting will be held on the 15th of July at 5pm UTC.

This is a bit short notice, but if there are any topics you would like to see discussed during this meeting, this is the discussion to post them on !

**Comment by joshka** [comment]
I'm 50/50 on whether I will be able to make it, but here's a few points of discussion

- New release 👍
  - definitely due for this ASAP and we're in a good place to do so soon
- Nightly releases 👍 
  - Why? Because it means that users can publish crates that rely on beta changes
  - How? GitHub scheduled workflow to release a v0.x.y-alpha.1 See https://doc.rust-lang.org/cargo/reference/resolver.html#pre-releases
  - ❓any positive / negative perspective on pre-release crates?
- Maintainers:
  - 👍 to needing more.
  - 👀 Most of all we need people to review code
  - 🥷 reviews are possible without being a maintainer, a good metric for becoming a maintainer is fake it until you make it.
- Reaching out to tui-rs / fdehau to take over tui-rs
  - why? existing tui-rs users helping shape the direction of ratatui would be useful
  - How?
    - move the tui-rs project under tui-rs-revival and archive
    - move all the open issues from tui-rs to ratatui and triage
    - release a compatible tui-rs changeover version
- ❓what is the project goal / direction
  - noting the existing code / backwards compatibility etc.
  - My general approach is to break as little as possible without prior notification (i.e. deprecate first then break)
- :book: ratatui-book

**Comment by mindoodoo** [comment]
Two small things on my side :

- Potentially adopting a no force push policy (issues seeing diff from last review)
- Add doc comments even if they may seem redundant to method name (usefull for new rust users)

**Reply by joshka** [reply]
We need a really good doc example and then make that the bar to strive for.

**Comment by orhun** [comment]
Reiterating a bit but I would be happy to discuss the following topics:
- New release (Is there any pending PRs that we are planning to incorporate?)
- Nightly releases (#147 - I can work on it in the following days if agreed.)
- New maintainers
  - Let's have more people to review and increase the required review count to 3 for better collaboration.
  - Do we have correct permissions set up for newly added maintainers?
- Discord
  - Do we need more/less roles? Is there anything needed on that side? (e.g. cleanup)
- Website
  - What do you think of having e.g. `ratatui.rs` as a landing page/documentation?
- Fun stuff
  - [Charm](https://charm.sh/) of Rust (cc @joshka)

**Reply by joshka** [reply]
The main con on having more required reviewers is the amount of wip increases too much. Strong opinion that this should be 1 or zero with perhaps a second reviewer when the code is gnarly.

**Reply by joshka** [reply]
Also. We should drop canvas::* from the prelude as Line clashes and anyone can import canvas::* just as easy.

**Reply by mindoodoo** [reply]
I personally wouldn't mind 3 reviewers required, but I see @joshka's point.

I think a sound compromise would be coming back to 2 reviewers, or an alternative which I think might be worth discussing would be to keep it at one **but** only for convenience on small PRs (lint issues, docs, **small** bug fixes, etc...) and make it a rule to wait for 2 review for anything larger such as new features, bigger bug fixes and what not.

**Reply by joshka** [reply]
Yeah - hard 1, soft 2 would be fine

**Reply by joshka** [reply]
> What do you think of having e.g. ratatui.rs as a landing page/documentation?
Let's get it!

**Comment by stappersg** [comment]
Idea: Some kind of schedule for these meetings.

Like _each six weeks_  or _third saturday of the month_.


(No, [I](https://github.com/tui-rs-revival/rust-tui-template/commits?author=stappersg) can't make it today )

**Reply by mindoodoo** [reply]
Great idea, I think users being able to look forward to a meeting might be interesting !
